src/Registry/TypeRegistry.php

```diff
@@ -661,6 +661,12 @@ protected function register_connections_from_config( array $config ) {
 	 * @return void
 	 */
 	public function register_type( string $type_name, $config ): void {
+
+		/**
+		* Decodes the type name, so it works with different languages (e.g. Chinese, Hebrew, etc.)
+		*/
+		$type_name = urldecode( $type_name );
+
```

Comment:
@scottyzen can you explain the reason/case for this change?

GraphQL type names can only be English alphanumeric (http://spec.graphql.org/draft/#sec-Names), and I'm pretty sure that by this point in the lifecycle `graphql_format_type_name()` has already run, stripping out any non A-Z.0-9  characters (including the `%` markers which would break `urldecode()` ).

Is there a scenario where either the encoded or decoded string is going to be all (english) alpha-numeric characters?

(ps: @jasonbahl , maybe we should expand `is_valid_graphql_name()` to check for more than a leading `int`? I'm not seeing obvious negative side effects, and we might be able to catch some currently fatal errors. )

Reply:
Hi @justlevine. There may be a better workaround to be done. But for some languages, the type doesn't get registered. For example, in Hebrew, the type name is "הטבה לרוכשים" and it doesn't work. So I encoded it and it worked. I'm not sure if it's the best solution though. Maybe the a **A-Z.0-9** needs to be expanded to handle other characters like Hebrew, Arabic, etc. It's just something I came across recently while working on a that had a lot of Hebrew content.

Reply:
> it's just something I came across
@scottyzen can you share more about the specific path that was registering a type with hebrew characters?

was it a direct call to `register_graphql_type()` or something like a page template that the Schema tries to register types from? 

Would be curious to learn more about what exactly was registering the type.

Reply:
Hey guys, so I think I found the issue. I'll try to explain it as best I can. Say I go to create 2 new Product Attributes.
- First one named: test-אחד 
- Second one named: test-שני .

From the GraphQL IDE, I get the following response:

```js
{
  "productAttributes": {
    "nodes": [
      {
        "name": "test-אחד"
      },
      {
        "name": "test-שני"
      }
    ]
  }
}
```

That response is correct. But because the register_type() only allows A-Z.0-9, it strips out other characters ( in this case the Hebrew characters) and the type name becomes "test-". So it fails when I try to create the second attribute because the type name is the same.
```js
{
    "type": "DUPLICATE_TYPE",
    "message": "You cannot register duplicate Types to the Schema. The Type 'ContentTypesOfPaTestEnum' already exists in the Schema. Make sure to give new Types a unique name.",
    "type_name": "ContentTypesOfPaTestEnum"
}
```

I hope this helps.

Reply:
Thanks @scottyzen .

That tracks with what I said above. You can see [this issue](https://github.com/wp-graphql/wp-graphql/issues/1658) for the same problem in a different context. Tl;Dr WPGraphQL should probably include some sort of transliteration function (e.g. via `iconv`), and in the interim you can use the built-in type name filters to str_replace `'א'=>'a'` etc.

In the context of this PR, I think you should revert this line, so we can add the necessary unit tests and get this merged (I'm happy to try and find time this week to write them up if you're unable).

Reply:
Great, thanks @justlevine.